### ISSUE TITLE
Convert ErgoTree Scans to Coll[Byte] 

### ISSUE THREAD

**Issue by SethDusek** [ergoplatform/oracle-core#83]
See: https://github.com/ergoplatform/ergo/issues/1781#issuecomment-1196702640 

Currently scans for contract address fail, because ergo node expects serialized Coll[Byte] in scans it seems. The easiest way to convert ErgoTree to this format would be something like:

```rust 
fn ergo_tree_to_scan_bytes(tree: &ErgoTree) -> String {
   base16::encode_lower(Constant::from(tree.sigma_serialize_bytes().unwrap()).sigma_serialize_bytes())
}
```

edit: I have a fix in my updatecontract branch, and I can confirm that the scans now work. Before ergo node was also throwing an exception in one of the scans.

**Comment by SethDusek** [comment]
Alternatively, node_interface::address_to_bytes might also work, but I have not tested it

**Comment by greenhat** [comment]
Oh wow, I was not expecting that. Good catch!

`node_interface::address_to_bytes` does it via node REST API call, so it's not advised.

**Comment by greenhat** [comment]
Would you consider implementing it via a trait:
```rust
pub trait ToScanBytes {
  fn to_scan_bytes(&self) -> String
}

impl ToScanBytes for ErgoTree {
  ...
}
```

I think it's more ergonomic than a function.

**Comment by SethDusek** [comment]
Fixed in #80
